src/components/common/Swap/components/CoinsListModal/CoinsListModal.tsx

```diff
@@ -1,25 +1,39 @@
+"use client";
 import SearchIcon from "@/src/components/icons/Search/SearchIcon";
 import CoinListItem from "@/src/components/common/Swap/components/CoinListItem/CoinListItem";
 import {ChangeEvent, memo, useEffect, useMemo, useRef, useState} from "react";
 import styles from "./CoinsListModal.module.css";
 import {BN, CoinQuantity} from "fuels";
-import { useAssetList } from "@/src/hooks/useAssetList";
+import {useAssetList} from "@/src/hooks/useAssetList";
 import UnknownCoinListItem from "../UnknownCoinListItem";
+import {useQuery} from "@tanstack/react-query";
+import {VerifiedAssets} from "../CoinListItem/checkIfCoinVerified";
 
 type Props = {
   selectCoin: (assetId: string | null) => void;
   balances: CoinQuantity[] | undefined;
   verifiedAssetsOnly?: boolean;
 };
 
-const priorityOrder: string[] = ['ETH', 'USDC', 'USDT', 'FUEL'];
-const lowPriorityOrder: string[] = ['DUCKY'];
+const priorityOrder: string[] = ["ETH", "USDC", "USDT", "FUEL"];
+const lowPriorityOrder: string[] = ["DUCKY"];
 
 const assetIdRegex = /^0x[0-9a-fA-F]{64}$/;
 
-const CoinsListModal = ({ selectCoin, balances, verifiedAssetsOnly }: Props) => {
-  const { assets, isLoading } = useAssetList();
-  const [value, setValue] = useState('');
+const CoinsListModal = ({selectCoin, balances, verifiedAssetsOnly}: Props) => {
+  const {data: verifiedAssetData} = useQuery({
```

Comment:
Can we move this into it's own hook (`useVerifiedAssets`)?

Reply:
Done!